packages/api/tests/lambdas/test-write-db-dlq-records-to-s3.js

```diff
@@ -109,14 +109,233 @@ test.serial('write-db-dlq-records-to-s3 throws error if system bucket is not def
+        error: 'anError',
+        detail: {
+          status: 'RUNNING',
+          input: JSON.stringify({
+            meta: { collection: { name: 'aName' } },
+            payload: { granules: [{ a: 'b' }, { granuleId: 'b' }] },
+          }),
+          executionArn: 'execArn',
+          stateMachineArn: 'SMArn',
+        },
+        collection: 'aName',
+        granules: ['unknown', 'b'],
+        execution: 'execArn',
+        stateMachine: 'SMArn',
+      },
+    }, {
+      mangled: {
+        error: 'anError',
+        detail: {
+          status: 'SUCCEEDED',
+          output: JSON.stringify({
+            meta: { collection: { name: 'aName' } },
+            payload: { granules: [{ granuleId: 'b' }] },
+          }),
+          executionArn: 'execArn',
+          stateMachineArn: 'SMArn',
+        },
+      },
+      expected: {
+        error: 'anError',
+        detail: {
+          status: 'SUCCEEDED',
+          output: JSON.stringify({
+            meta: { collection: { name: 'aName' } },
+            payload: { granules: [{ granuleId: 'b' }] },
+          }),
+          executionArn: 'execArn',
+          stateMachineArn: 'SMArn',
+        },
+        collection: 'aName',
+        granules: ['b'],
+        execution: 'execArn',
+        stateMachine: 'SMArn',
+      },
+    }, {
+      mangled: {
+        error: 'anError',
+        detail: {
+          status: 'SUCCEEDED',
+          output: JSON.stringify({
+            meta: { collection: { name: 'aName' } },
+            payload: { granules: 'a' },
+          }),
+          executionArn: 'execArn',
+          stateMachineArn: 'SMArn',
+        },
+      },
+      expected: {
+        error: 'anError',
+        detail: {
+          status: 'SUCCEEDED',
+          output: JSON.stringify({
+            meta: { collection: { name: 'aName' } },
+            payload: { granules: 'a' },
+          }),
+          executionArn: 'execArn',
+          stateMachineArn: 'SMArn',
+        },
+        collection: 'aName',
+        granules: 'unknown',
+        execution: 'execArn',
+        stateMachine: 'SMArn',
+      },
+    }, {
+      mangled: {
+        error: 'anError',
+        detail: {
+          status: 'SUCCEEDED',
+          output: JSON.stringify({
+            meta: { collection: { name: 'aName' } },
+            payload: { granules: [{ granuleId: 'a' }, { granuleId: 'b' }] },
+          }),
+          executionArn: 'execArn',
+        },
+      },
+      expected: {
+        error: 'anError',
+        detail: {
+          status: 'SUCCEEDED',
+          output: JSON.stringify({
+            meta: { collection: { name: 'aName' } },
+            payload: { granules: [{ granuleId: 'a' }, { granuleId: 'b' }] },
+          }),
+          executionArn: 'execArn',
+        },
+        collection: 'aName',
+        granules: ['a', 'b'],
+        execution: 'execArn',
+        stateMachine: 'unknown',
+      },
+    },
+  ];
+  for (let i = 0; i < messages.length; i += 1) {
```

Comment:
Nit: consider removing iterator
```suggestion
  const results = await Promise.all(
    messages.map((message) => formatCumulusDLAObject(message.mangled))
  );
  results.foreach((result, index) => {
    t.deepEqual(result, messages[index].expected);
  });
});

```